nominal/config.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import annotations
+
+import logging
+from dataclasses import asdict, dataclass
+from pathlib import Path
+from typing import Literal, Mapping
+
+import yaml
+from typing_extensions import Self  # typing.Self in 3.11+
+
+from nominal._config import _DEFAULT_NOMINAL_CONFIG_PATH as DEPRECATED_NOMINAL_CONFIG_PATH
+from nominal.exceptions import NominalConfigError
+
+DEFAULT_NOMINAL_CONFIG_PATH = Path("~/.config/nominal/config.yml").expanduser().resolve()
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class ConfigProfile:
+    base_url: str
+    token: str
+    workspace_rid: str | None = None
+
+
+@dataclass(frozen=True)
+class NominalConfig:
+    """Nominal configuration. Stores connection profiles.
+
+    By default, the configuration is stored in ~/.config/nominal/config.yml.
+
+    Example config file
+    -------------------
+    version: 2
+    profiles:
+      default:
+        base_url: https://api.gov.nominal.io/api
+        token: nominal_api_key_...
+      staging:
+        base_url: https://api-staging.gov.nominal.io/api
+        token: eyJ...
+        workspace_rid: ri.security.gov-staging.workspace.82db1f3a-568e-418e-a2d0-0575396f29a2
+      dev:
+        base_url: https://api.nominal.test
+        token: eyJ...
+
+
+    For production environments, the typical URL is: https://api.gov.nominal.io/api
+    For staging environments, the typical URL is: https://api-staging.gov.nominal.io/api
+    For local development, the typical URL is: https://api.nominal.test (note the lack of the /api suffix)
+    """
+
+    profiles: Mapping[str, ConfigProfile]
+    version: Literal[2]
```

Comment:
Do you want to move this up by one line, so that the config file starts with `version: 2`? Right now, the profiles are written out first.

Reply:
🤷🏼 idk if it matters much either way, i think i went back and forth on defaulting this param in the first place